metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py

```diff
@@ -513,7 +573,8 @@ def generate_dataset_profile(  # noqa: C901 (complexity)
                 columns_profiling_queue.append(column_spec)
 
                 self._get_column_type(column_spec, column)
-                self._get_column_cardinality(column_spec, column)
+                if column_spec.type_ != ProfilerDataType.UNKNOWN:
+                    self._get_column_cardinality(column_spec, column)
```

Comment:
Unfortunately `self._get_column_type` can be executed asynchronously, so it's not guaranteed that `column_spec.type_` will actually be set here.

Reply:
I think addressed this as well.